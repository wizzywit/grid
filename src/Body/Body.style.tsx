import styled from "styled-components";

export const Styles = {
  Container: styled.div`
    background-color: ${({ theme }) => theme.colors.main};
    padding: 50px 200px 50px 200px;
    min-height: calc(100vh - 188px);

    @media only screen and (max-width: 800px) {
      padding: 25px;
    }

    .filters {
      color: ${({ theme }) => theme.colors.title};
      font-weight: bold;
    }
  `,
  ShapesContainer: styled.div`
    display: flex;
    align-items: center;
    /* justify-content: space-between;
    width: 50%; */
    gap: 25px;
    margin-bottom: 50px;
  `,
};
