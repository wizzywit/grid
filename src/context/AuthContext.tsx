import React from "react";

interface DataArray {
  color: string;
  shape: string;
}

export type Filter = (
  c: Array<string> | ((c: Array<string>) => Array<string>)
) => void;

type PerformFilterType = (
  shape: string,
  shapesFilter: Array<string> | undefined,
  filter: Filter | undefined,
  type: string
) => void;

type AuthContextType = {
  login?: () => void;
  logout?: () => void;
  data?: Array<DataArray>;
  shapesFilter?: Array<string>;
  setShapesFilter?: Filter;
  colorsFilter?: Array<string>;
  setColorsFilter?: Filter;
  isLoggedIn?: boolean;
  performSelect?: PerformFilterType;
};
const AuthContext = React.createContext<AuthContextType>({});

export default AuthContext;
