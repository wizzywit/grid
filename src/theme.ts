import { DefaultTheme } from "styled-components";

const myTheme: DefaultTheme = {
  borderRadius: "5px",

  colors: {
    main: "#f5f6fc",
    secondary: "white",
    title: "#5b95f7",
  },
};

export { myTheme };
