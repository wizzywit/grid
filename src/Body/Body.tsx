import React, { useContext } from "react";
import { Styles } from "./Body.style";
import Result from "../components/Result/Result";
import Shapes from "../components/Shapes/Shapes";
import Colors from "../components/Colors/Colors";
import AuthContext from "../context/AuthContext";
import { gridTitle } from "../helper";

const Body = () => {
  const { colorsFilter, shapesFilter, data } = useContext(AuthContext);
  const title =
    colorsFilter && shapesFilter ? gridTitle(colorsFilter, shapesFilter) : "";
  return (
    <Styles.Container>
      <h2 data-testid="title">Filters</h2>
      <p className="filters" data-testid="shape-title">
        Shapes
      </p>
      <Shapes />
      <p className="filters" data-testid="color-title">
        Colors
      </p>
      <Colors />
      <div className="result">
        <h2>{title}</h2>
        <p>({data?.length})</p>
      </div>
      <Result />
    </Styles.Container>
  );
};

export default Body;
