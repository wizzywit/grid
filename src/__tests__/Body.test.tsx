import React from "react";

import { render } from "@testing-library/react";

import Body from "../Body/Body";
import AuthContext from "../context/AuthContext";
import { ThemeProvider } from "styled-components";
import { myTheme } from "../theme";

const data = [
  { shape: "oval", color: "red" },
  { shape: "traingle", color: "grey" },
  { shape: "round", color: "blue" },
  { shape: "square", color: "green" },
  { shape: "rectangle", color: "grey" },
];

describe("Body", () => {
  test("should display body containing filters and results", async () => {
    const { container, findAllByTestId, findByTestId } = render(
      <AuthContext.Provider value={{ data }}>
        <ThemeProvider theme={myTheme}>
          <Body />
        </ThemeProvider>
      </AuthContext.Provider>
    );
    expect(container).toMatchSnapshot();
    const children = await findAllByTestId("item");
    const title = await findByTestId("title");
    const colorsTitle = await findByTestId("color-title");
    const shapesTitle = await findByTestId("shape-title");

    expect(title).toHaveTextContent("Filters");
    expect(shapesTitle).toHaveTextContent("Shapes");
    expect(colorsTitle).toHaveTextContent("Colors");
    expect(children).toHaveLength(data.length);
  });
});
