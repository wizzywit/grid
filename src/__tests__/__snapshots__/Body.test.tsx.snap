// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should display body containing filters and results 1`] = `
<div>
  <div
    class="sc-bdnxRM dWjOYm"
  >
    <h2
      data-testid="title"
    >
      Filters
    </h2>
    <p
      class="filters"
      data-testid="shape-title"
    >
      Shapes
    </p>
    <div
      class="sc-hKFxyN clYRje"
    />
    <p
      class="filters"
      data-testid="color-title"
    >
      Colors
    </p>
    <div
      class="sc-jSFjdj ghaAYX"
    />
    <div
      class="result"
    >
      <h2 />
      <p>
        (
        5
        )
      </p>
    </div>
    <div
      class="sc-gtsrHT ktoWBZ"
    >
      <div
        class="sc-dlnjwi bKRnTG"
        color="red"
        data-testid="item"
      >
        <div
          class="oval"
          data-shape="oval"
          data-testid="shape"
        />
      </div>
      <div
        class="sc-dlnjwi iCSYJP"
        color="grey"
        data-testid="item"
      >
        <div
          class="traingle"
          data-shape="traingle"
          data-testid="shape"
        />
      </div>
      <div
        class="sc-dlnjwi cfJJck"
        color="blue"
        data-testid="item"
      >
        <div
          class="round"
          data-shape="round"
          data-testid="shape"
        />
      </div>
      <div
        class="sc-dlnjwi ebcTyC"
        color="green"
        data-testid="item"
      >
        <div
          class="square"
          data-shape="square"
          data-testid="shape"
        />
      </div>
      <div
        class="sc-dlnjwi iCSYJP"
        color="grey"
        data-testid="item"
      >
        <div
          class="rectangle"
          data-shape="rectangle"
          data-testid="shape"
        />
      </div>
    </div>
  </div>
</div>
`;
